packages/cxl-jwplayer/src/components/cxl-jwplayer.js

```diff
@@ -1,17 +1,152 @@
+  player = null;
+
+  @property({ type: Object })
+  config = {};
+
+  @property({ type: String })
+  mediaId = '';
+
+  constructor() {
+    super();
+    this.jwplayer = window.jwplayer || null;
+    if (!this.jwplayer) console.error('JWPlayer library not found.');
+  }
+
+  /*
+   * Implemented to avoid default, which attaches a shadowDOM tree byt default.
+   */
+  createRenderRoot() {
+    return this;
   }
 
   render() {
-    return html``;
+    return html`
+      <h2>cxl-jwplayer</h2>
+      <div id="cxl-jwplayer-container" class="cxl-jwplayer-container">
+        <div id="cxl-jwplayer" class="cxl-jwplayer jwplayer"></div>
+        ${this.config.transcript
+          ? html`
+              <div id="cxl-transcript-container" class="transcript-container">
+                <div id="searchbox" class="searchbox">
+                  <input
+                    id="search"
+                    type="search"
+                    class="search"
+                    autocomplete="off"
+                    results="5"
+                    placeholder="Type and press enter to search"
+                  />
+                  <span id="match" class="match hidden">0 of 0</span>
+                </div>
+                <div class="jw-aspect jw-reset"></div>
+                <div class="transcript-wrapper">
+                  <div id="transcript" class="transcript"></div>
+                </div>
+              </div>
+            `
+          : ''}
+      </div>
+    `;
+  }
+
+  firstUpdated() {
+    // Run player setup if minimal requirements are met
+    // Bail if jwplayer is not loaded or mediaId is not set
+    if (!this.jwplayer || !this.mediaId) return;
+
+    this.setupPlayer();
+  }
+
+  setupPlayer() {
+    /**
+     *
+     * Configure and Setup the Player
+     *
+     */
+    // Fetched via GET request to JW Platform
+    // One video is an object. Multiple videos is an array of objects.
+    // Paths are hard coded for now, while developing this implementation.
+    const data = {
+      config: {
+        playlist: [
+          {
+            file: `https://cdn.jwplayer.com/manifests/${this.mediaId}.m3u8`,
+            image: `https://cdn.jwplayer.com/thumbs/${this.mediaId}-720.jpg`,
+            tracks: [
+              {
+                file: `https://cdn.jwplayer.com/strips/${this.mediaId}-120.vtt`,
+                kind: 'thumbnails'
+              },
+              {
+                file: 'https://assets-jpcust.jwpsrv.com/tracks/HujCQO8E',
+                kind: 'captions',
+                label: 'English',
+                language: 'en_US'
+              },
+              {
+                file: 'https://content.jwplatform.com/tracks/qb36nsWb.vtt', // need to be able to set these dynamically...
+                kind: 'chapters'
+              }
+            ]
+          }
+        ]
+      }
+    };
+
+    const { config } = data;
+    config.plugins = {};
+
+    /**
+     *
+     * Conditionally load plugins
```

Comment:
See comment above for how plugin loading is working and if it will still work for components.